oecophylla/util/parse.py

```diff
@@ -0,0 +1,149 @@
+import re
+import pandas as pd
+
+# Option 1 : parse samples from directory
+
+def parse_filenames_to_df(filepaths, pattern = None, column_names = None):
+    """ Generates sample sheet from filenames using a regex pattern.
+
+    Parameters
+    ----------
+    filepaths: list of str
+        List of file paths to parse
+    pattern: str
+        Regex pattern to recognize the filenames, to load into the
+        sample sheet.
+        default: '^((.+?)_(S\d+)_(L\d+)_(R[12])_(\d+)\.(.+))$'
+    column_names: list
+        Column names for the generated sample sheet.  Note that this
+        is assumed to be in the same ordering as the regex pattern.
+        default: ['File', 'Sample', 'Index', 'Lane',
+                  'Read', 'Run', 'Extension']
+
+
+    """
+    if pattern is None:
+        pattern = '^((.+?)_(S\d+)_(L\d+)_(R[12])_(\d+)\.(.+))$'
+    if column_names is None:
+        column_names = ['File', 'Sample', 'Index', 'Lane',
```

Comment:
it looks like downstream code (e.g., `get_sample_reads_df` assumes at least some of these columns exist. Is there a usecase for the column names being parameterized (same q for the pattern)?

Reply:
Done

Reply:
This was done in the inverse way to what I was thinking... will `sample_col` in `extract_sample_reads` ever not be `Sample`? Perhaps not worth blocking though